exporter/elasticsearchexporter/integrationtest/config.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package integrationtest
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/knadh/koanf/providers/confmap"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/v2"
+	"github.com/stretchr/testify/require"
+)
+
+type config struct {
+	GRPCEndpoint string `koanf:"grpc_endpoint"`
+	ESLogsIndex  string `koanf:"es_logs_index"`
+	Debug        bool   `koanf:"debug"`
+}
+
+func loadConfig(t testing.TB) config {
+	k := koanf.New(".")
+	defaultConfProvider := confmap.Provider(map[string]interface{}{
+		"grpc_endpoint": "127.0.0.1:4317",
+		"debug":         false,
+		"es_logs_index": "esexportertest",
+	}, ".")
+	if err := k.Load(defaultConfProvider, nil); err != nil {
+		require.NoError(t, err, "failed to load default config")
+		return config{}
+	}
+	if err := k.Load(env.Provider("", ".", func(s string) string {
```

Comment:
question: what's the reason to use a config lib and load config from env? What's the use case for configurable cfg, given that runner is already configurable with restartCollector and mockESFailure? Just wanted to check if the benefit is worth the extra dependency

Reply:
Ah, good catch. The configurations were added to support configuring real Elasticsearch instances by overriding the ES configurations however, I decided to drop that support in favor of mock. I think the config is no longer required, the only required config is `Debug` which we can get directly from env. Let me make the changes here.

Reply:
Fixed via https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31848/commits/e44221077afc3b75efa1e2d0023c0cab508b3b5f